ui/src/shared/reducers/sources.js

```diff
@@ -0,0 +1,29 @@
+export default function sources(state = [], action) {
+  switch (action.type) {
+    case 'LOAD_SOURCES': {
+      const updatedSources = action.payload.sources;
+      return updatedSources;
+    }
+
+    case 'SOURCE_UPDATED': {
+      const {source} = action.payload;
+      const updatedIndex = state.findIndex((s) => s.id === source.id);
+      const updatedSources = Object.assign({}, state);
+      updatedSources[updatedIndex] = source;
```

Comment:
`Object.assign` is great in reducers - good readability once you understand it, and avoids mutations.